cmd/soroban-rpc/internal/db/db.go

```diff
@@ -52,7 +53,14 @@ type dbCache struct {
 
 type DB struct {
 	db.SessionInterface
-	cache dbCache
+	cache *dbCache
+}
+
+func (db *DB) Clone() *DB {
+	return &DB{
+		SessionInterface: db.SessionInterface.Clone(),
+		cache:            db.cache,
```

Comment:
I'm not sure if the cache should be propagated to the cloned `DB` instance since the cache contains data which may be bound to the current running transaction. If you look at the implementation of `Clone()` on the db session you will find that it does not clone the transaction state:

https://github.com/stellar/go/blob/master/support/db/session.go#L120-L127

Reply:
Uhm, I need to think about this. I think it's probably fine to clone it at selected places. I am tempted to remove the method and just copy it manually in the only `Clone()` invokation

Reply:
I just removed the Clone() method